File: hello_world/src/hello_world_mpi.f90
PROGRAM hello_f
  USE mpi
  IMPLICIT NONE
  INTEGER :: ierr
  INTEGER :: thistask
  INTEGER :: ntask
  
  CALL mpi_init(ierr)
  CALL mpi_comm_rank(MPI_COMM_WORLD, thistask, ierr)
  CALL mpi_comm_size(MPI_COMM_WORLD, ntask, ierr)
  
  WRITE (*,"(A,I0,A,I0)") "Hello, World! from task ", thistask, " of ",ntask 
  
  CALL mpi_finalize(ierr)
END PROGRAM hello_f

File: hello_world/src/hello_world_mpi.c
#include <stdio.h>
#include <stdlib.h>
#include <mpi.h>

int main (int argc, char *argv[])
{
  int ThisTask;
  int NTask;
  
  MPI_Init(&argc, &argv);
  MPI_Comm_rank(MPI_COMM_WORLD, &ThisTask);
  MPI_Comm_size(MPI_COMM_WORLD, &NTask);
  printf("Hello, World! from Task %d of %d\n", ThisTask, NTask);
  MPI_Barrier(MPI_COMM_WORLD);
  MPI_Finalize();
  return EXIT_SUCCESS;
}
